src/Interpreters/Streaming/Aggregator.cpp

```diff
@@ -103,6 +107,73 @@ Columns materializeKeyColumns(Columns & columns, ColumnRawPtrs & key_columns, co
 
     return materialized_columns;
 }
+
+Arena * getArena(AggregatedDataVariants & variants, AggregateStateType type)
```

Comment:
`getArena` can be a member function of `AggregatedDataVariants` and we may use `getPool()` for convention